File: local_examples/cmds_string/redis-py/cmds_string.py
# EXAMPLE: cmds_string
# HIDE_START
import redis

r = redis.Redis(decode_responses=True)
# HIDE_END

# REMOVE_START
r.delete("key1", "key2", "mykey", "nonexisting")
# REMOVE_END

# STEP_START mget
r.set("key1", "Hello")
r.set("key2", "World")

mget_result = r.mget("key1", "key2", "nonexisting")
print(mget_result)
# >>> ['Hello', 'World', None]
# STEP_END

# REMOVE_START
assert mget_result == ["Hello", "World", None]
r.delete("key1", "key2", "nonexisting")
# REMOVE_END

# STEP_START incr
incr_result1 = r.set("mykey", "10")
print(incr_result1)
# >>> True

incr_result2 = r.incr("mykey")
print(incr_result2)
# >>> 11

incr_result3 = r.get("mykey")
print(incr_result3)
# >>> 11
# STEP_END

# REMOVE_START
assert incr_result1 is True
assert incr_result2 == 11
assert incr_result3 == "11"
r.delete("mykey")
# REMOVE_END

File: local_examples/cmds_string/ioredis/cmds-string.js
// EXAMPLE: cmds_string

// HIDE_START
import assert from 'node:assert';
import { Redis } from 'ioredis';

const redis = new Redis();
// HIDE_END

// REMOVE_START
await redis.del('key1', 'key2', 'mykey', 'nonexisting');
// REMOVE_END

// STEP_START mget
await redis.set('key1', 'Hello');
await redis.set('key2', 'World');

const mgetResult = await redis.mget('key1', 'key2', 'nonexisting');
console.log(mgetResult); // >>> ['Hello', 'World', null]
// STEP_END

// REMOVE_START
assert.deepEqual(mgetResult, ['Hello', 'World', null]);
await redis.del('key1', 'key2', 'nonexisting');
// REMOVE_END

// STEP_START incr
const incrResult1 = await redis.set('mykey', '10');
console.log(incrResult1); // >>> OK

const incrResult2 = await redis.incr('mykey');
console.log(incrResult2); // >>> 11

const incrResult3 = await redis.get('mykey');
console.log(incrResult3); // >>> 11
// STEP_END

// REMOVE_START
assert.equal(incrResult1, 'OK');
assert.equal(incrResult2, 11);
assert.equal(incrResult3, '11');
await redis.del('mykey');
// REMOVE_END

// HIDE_START
redis.disconnect();
// HIDE_END
